src/framework/ui/internal/navigationcontroller.cpp

```diff
@@ -1198,6 +1199,42 @@ void NavigationController::doTriggerControl()
     activeCtrl->trigger();
 }
 
+void NavigationController::onConfirm()
+{
+    TRACEFUNC;
+    MYLOG() << "====";
+    INavigationSection* activeSec = activeSection();
+    if (!activeSec) {
+        return;
+    }
+
+    INavigation::EventPtr e = Event::make(Event::Confirm);
+    activeSec->onEvent(e);
+    if (e->accepted) {
```

Comment:
We might want to add logic in `AbstractNavigation::onEvent` to make `Confirm` events non-accepted by default; I think most sections won't handle these events, in which case it seems good to me to send a Trigger event to the active control instead (to preserve the old behaviour where possible).